security/pkg/pki/ca/ca.go

```diff
@@ -183,11 +183,15 @@ func NewSelfSignedIstioCAOptions(ctx context.Context,
 		// Write the key/cert back to secret so they will be persistent when CA restarts.
 		secret := k8ssecret.BuildSecret("", CASecret, namespace, nil, nil, nil, pemCert, pemKey, istioCASecretType)
 		if _, err = client.Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
-			pkiCaLog.Errorf("Failed to write secret to CA (error: %s). Abort.", err)
-			return nil, fmt.Errorf("failed to create CA due to secret write error")
+			// Ignore error if secret got created by another replica
+			if caSecret, scrtErr = client.Secrets(namespace).Get(context.TODO(), CASecret, metav1.GetOptions{}); scrtErr != nil {
```

Comment:
One replica fails at writing a CA secret does not necessarily mean the Get(CASecret) will succeed (e.g., it may take a while for the CA secret writing from another replica to complete). Thus, the Get(CASecret) here may not always work.

Reply:
Would we get write failures for "already exists", when the other write was in progress? Or only once completed? If so, checking for specific error "secret already exists" and doing a read only in that case helps?

Reply:
It is hard to verify the correctness of code behavior under concurrent writes/reads. 

The existing implementation is simple and easy to analyze its correctness, which returns and retries if writing CA secret fails.I think the behavior of existing implementation is fine: the error log is harmless and may be improved by changing the log type from "error" to "info".

The PR has a more complicated implementation than the existing implementation and its behavior under concurrent writes/reads is more difficult to verify. Moreover, the existing implementation is stable and battle-tested.

Reply:
@lei-tang Understand your concern. I have anyways updated the PR adding additional logs, in case that makes the new implementation easier to debug in case of failures. Also fixed the logging inconsistency. Ofcourse retries are already there, which makes the problem less severe, but just that we can avoid an unnecessary retry.